Source/WebCore/dom/StaticRange.cpp

```diff
@@ -83,4 +83,23 @@ void StaticRange::visitNodesConcurrently(JSC::AbstractSlotVisitor& visitor) cons
     addWebCoreOpaqueRoot(visitor, end.container.get());
 }
 
+bool StaticRange::isValid() const
+{
+    Node& startContainer = this->startContainer();
+    Node& endContainer = this->endContainer();
+
+    if (collapsed())
+        return false;
+    if (endOffset() < startOffset() && startContainer.isSameNode(&endContainer))
```

Comment:
Just do: &startContainer == &endContainer
isSameNode only exists for web backwards compatibility reasons.